src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeData.java

```diff
@@ -0,0 +1,120 @@
+package org.broadinstitute.hellbender.tools.exome.sexgenotyper;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.IntStream;
+
+/**
+ * A class for handling sex genotypes and their likelihoods.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class SexGenotypeData {
+    /* mandatory */
+    private final String sampleName;
+    private final String sexGenotype;
+    private final boolean hasExtendedGenotypingInfo;
+
+    /* optional */
+    private final Map<String, Double> logLikelihoodMap;
+
+    /**
+     * Public constructor.
+     *
+     * @param sampleName string identifer of the sample
+     * @param sexGenotype string identifier for the sex genotype of the sample
+     * @param sexGenotypesList list of all sex genotypes
+     * @param logLikelihoods log likelihood of all sex genotypes in the same order as {@param sexGenotypesList}
+     */
+    public SexGenotypeData(@Nonnull final String sampleName,
+                           @Nonnull final String sexGenotype,
+                           @Nullable final List<String> sexGenotypesList,
+                           @Nullable final double[] logLikelihoods) {
+        this.sampleName = sampleName;
+        this.sexGenotype = sexGenotype;
+
+        if (sexGenotypesList == null || logLikelihoods == null) {
+            logLikelihoodMap = null;
+            hasExtendedGenotypingInfo = false;
+        } else {
+            ParamUtils.isPositive(sexGenotypesList.size(), "List of sex genotypes must be non-empty");
+            if (logLikelihoods.length != sexGenotypesList.size()) {
+                throw new UserException.BadInput("The size of genotype likelihood array must be equal to the number of" +
+                        " genotypes");
+            }
```

Comment:
Sure, but shall we agree that the current implementation does the job? :) a map has to be constructed somewhere anyway.